src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts

```diff
@@ -114,8 +113,9 @@ export const selectUserStakingOpportunitiesWithMetadataByFilter = createSelector
         if (!stakingOpportunitiesById[stakingId] || !userStakingOpportunity) return undefined
 
         const userStakingOpportunityWithMetadata = {
-          ...stakingOpportunitiesById[stakingId],
           ...userStakingOpportunity,
+          ...stakingOpportunitiesById[stakingId],
```

Comment:
The reorder is on purpose - if `userStakingOpportunity` was ever to expose an `id` property (currently `userStakingId` to make things more legible), we want to make sure `id` is the one from the metadata, not the one from the userData.